test-data/unit/check-based-union-join.test

```diff
@@ -0,0 +1,62 @@
+[case testSimpleUnionAtJoin]
+a = [1, ""]
+reveal_type(a)  # N: Revealed type is "list[int | str]"
+
+
+[case testSubtype]
+class A: pass
+class B(A): pass
+
+reveal_type([A(), B()])  # N: Revealed type is "list[__main__.A]"
+
+
+[case testConditionalExpression]
+class A: pass
+class B: pass
+
+res = A() if bool() else B()
```

Comment:
what happens with those sus duck types or whatever? like `int | float` or `int | bool` where mypy treats them as the same type

Reply:
`bool` is actually an `int` in real life, but for the duck ones (or 'promoted' as mypy calls it), they will turned into `float`.

I could interject here and make it a `float | int`, but maybe that's inconsistent?

Reply:
prob sus to change the behavior for this one case. as long as ur confident it works without having to add a test u can resolve this